DQMOffline/Trigger/plugins/METplusTrackMonitor.cc

```diff
@@ -0,0 +1,342 @@
+  iEvent.getByToken(metToken_, metHandle);
+  reco::CaloMET caloMet = metHandle->front();
+  if(!metSelection_(caloMet)) return;
+
+  float met = caloMet.pt();
+  float metPhi = caloMet.phi();
+
+  edm::Handle<reco::PFJetCollection> jetsHandle;
+  iEvent.getByToken(jetToken_, jetsHandle);
+  if(jetsHandle->size() < njets_) return;
+  std::vector<reco::PFJet> jets;
+  jets.clear();
+  for(auto const & j : *jetsHandle) {
+    if(jetSelection_(j)) jets.push_back(j);
+  }
+  if(jets.size() < njets_) return;
+  if(njets_ > 0 && jets.size() > 0 && fabs(jets[0].eta()) > leadJetEtaCut_) return;
+  float deltaphi_metjet1 = (jets.size() > 0) ? fabs(deltaPhi(caloMet.phi(), jets[0].phi())) : 10.0;
+
+  edm::Handle<reco::VertexCollection> primaryVertices;
+  iEvent.getByToken(vtxToken_, primaryVertices);
+  if(!primaryVertices->size()) return;
+  const reco::Vertex* pv = nullptr;
+  for(auto const& v: *primaryVertices) {
+    if (!vtxSelection_(v)) continue;
+    pv = &v;
+    break;
+  }
+  if(pv == nullptr) return;
+
+  edm::Handle<reco::MuonCollection> muonHandle;
+  iEvent.getByToken(muonToken_, muonHandle);
+  if(muonHandle->size() < nmuons_) return;
+  std::vector<reco::Muon> muons;
+  muons.clear();
+  for(auto const & m : *muonHandle) {
+    bool passTightID = muon::isTightMuon(m, *pv) &&
+                       m.innerTrack()->hitPattern().trackerLayersWithoutMeasurement(reco::HitPattern::MISSING_INNER_HITS) == 0 &&
+                       m.innerTrack()->hitPattern().trackerLayersWithoutMeasurement(reco::HitPattern::TRACK_HITS) == 0;
+    if(muonSelection_(m) && passTightID) muons.push_back(m);
+  }
+  if(muons.size() < nmuons_) return;
+
+  // Filling MET leg histograms (denominator)
+  metME_variableBinning_.denominator->Fill(met);
+  metPhiME_.denominator->Fill(metPhi);
+  deltaphimetj1ME_.denominator->Fill(deltaphi_metjet1);
+
+  int ls = iEvent.id().luminosityBlock();
+  metVsLS_.denominator->Fill(ls, met);
+
+  // Apply the selection for the MET leg numerator
+  edm::Handle<trigger::TriggerEvent> triggerSummary;
+  iEvent.getByToken(theTrigSummary_, triggerSummary);
+  if(!triggerSummary.isValid()) {
+    edm::LogError("METplusTrackMonitor") << "Invalid collection: TriggerSummary" << "\n";
+    return;
+  }
+
+  trigger::TriggerObject hltMet, hltMetClean;
+  bool passesHltMetFilter = getHLTObj(triggerSummary, hltMetTag_, hltMet);
+  bool passesHltMetCleanFilter = getHLTObj(triggerSummary, hltMetCleanTag_, hltMetClean);
+
+  if(!passesHltMetFilter || !passesHltMetCleanFilter) return;
+
+  // Filling MET leg histograms (numerator)
+  metME_variableBinning_.numerator->Fill(met);
+  metPhiME_.numerator->Fill(metPhi);
+  deltaphimetj1ME_.numerator->Fill(deltaphi_metjet1);
+  metVsLS_.numerator->Fill(ls, met);
+  metVsHltMet_.numerator->Fill(hltMet.pt(), met);
+  metVsHltMetClean_.numerator->Fill(hltMetClean.pt(), met);
+
+  // Filling track leg histograms (denominator)
+  double leadMuonPt = muons.size() ? muons[0].pt() : -1.0;
```

Comment:
Same as above, if you mean muons.clear() then okay, otherwise empty() just returns a bool.